src/probnum/prob/random_variable/_dirac.py

```diff
@@ -0,0 +1,154 @@
+from typing import Callable, Optional, TypeVar
+
+import numpy as np
+
+from probnum import utils as _utils
+from probnum.prob import _random_variable
+
+
+_ValueType = TypeVar("ValueType")
+
+
+class Dirac(_random_variable.RandomVariable[_ValueType]):
+    """
+    The Dirac delta distribution.
+
+    This distribution models a point mass and can be useful to represent
+    numbers as random variables with Dirac measure. It has the useful
+    property that arithmetic operations between a :class:`Dirac` random
+    variable and an arbitrary :class:`RandomVariable` acts in the same
+    way as the arithmetic operation with a constant.
+
+    Note, that a Dirac measure does not admit a probability density
+    function but can be viewed as a distribution (generalized function).
+
+    Parameters
+    ----------
+    support : scalar or array-like or LinearOperator
+        The support of the dirac delta function.
+
+    See Also
+    --------
+    RandomVariable : Class representing general random variables.
+
+    Examples
+    --------
+    >>> from probnum.prob import random_variable as rv
+    >>> rv1 = rv.Dirac(support=0.)
+    >>> rv2 = rv.Dirac(support=1.)
+    >>> rv = rv1 + rv2
+    >>> rv.sample(size=5)
+    array([1., 1., 1., 1., 1.])
+    """
+
+    def __init__(
+        self,
+        support: _ValueType,
+        random_state: Optional[_random_variable.RandomStateType] = None,
+    ):
+        if np.isscalar(support):
+            support = _utils.as_numpy_scalar(support)
+
+        self._support = support
+
+        super().__init__(
+            shape=support.shape,
+            dtype=support.dtype,
+            random_state=random_state,
+            parameters={"support": support},
+            sample=self._sample,
+            in_support=lambda x: np.all(x == support),
+            cdf=lambda x: 0.0 if np.any(x < self._support) else 0.0,
+            cov=lambda: np.zeros_like(  # pylint: disable=unexpected-keyword-arg
+                self._support,
+                shape=(
+                    (self._support.size, self._support.size)
+                    if self._support.ndim > 0
+                    else ()
+                ),
+            ),
+            var=lambda: np.zeros_like(self._support),
+            properties={
+                "mode": self._support,  # TODO: Is this correct? Diracs don't have a pdf
```

Comment:
I think this is sensible _definition_ if one interprets the dirac delta function as the limit of a Gaussian pdf.

Reply:
Now that `Dirac`s are `DiscreteRandomVariables`, we should not need to define this anymore, since this is also the mode of the `pmf`, right?